docs/source/deepspeed.mdx

```diff
@@ -0,0 +1,542 @@
+--report_to "wandb"\
+```
+
+**Important code changes when using DeepSpeed Config File**
+
+1. DeepSpeed Optimizers and Schedulers. For more information on these, 
+see the [DeepSpeed Optimizers](https://deepspeed.readthedocs.io/en/latest/optimizers.html) and [DeepSpeed Schedulers](https://deepspeed.readthedocs.io/en/latest/schedulers.html) documentation.
+We will look at the changes needed in the code when using these.
+   
+   a. DS Optim + DS Scheduler: The case when both `optimizer` and `scheduler` keys present in the DeepSpeed config file.
+   In this situation, those will be used and user has to use `accelerate.utils.DummyOptim` and `accelerate.utils.DummyScheduler` to replace the PyTorch/Custom optimizers and schedulers in their code.
+   Below is the snippet from `examples/by_feature/deepspeed_with_config_support.py` showing this:
+   ```python
+   # Creates Dummy Optimizer if `optimizer` was spcified in the config file else creates Adam Optimizer
+   if "optimizer" not in accelerator.state.deepspeed_plugin.deepspeed_config:
+       optimizer = torch.optim.AdamW(optimizer_grouped_parameters, lr=args.learning_rate)
+   else:
+       optimizer = DummyOptim(param_groups=optimizer_grouped_parameters)
+
+   # Creates Dummy Optimizer if `optimizer` was spcified in the config file else creates Adam Optimizer
+   if "scheduler" not in accelerator.state.deepspeed_plugin.deepspeed_config:
+       lr_scheduler = get_scheduler(
+           name=args.lr_scheduler_type,
+           optimizer=optimizer,
+           num_warmup_steps=args.num_warmup_steps,
+           num_training_steps=args.max_train_steps,
+       )
+   else:
+       lr_scheduler = DummyScheduler()
+   ```
+   b. Custom Optim + Custom Scheduler: The case when both `optimizer` and `scheduler` keys are absent in the DeepSpeed config file.
+   In this situation, no code changes are needed from the user and this is the case when using integration via DeepSpeed Plugin.
+   In the above example we can see that the code reamins unchanged if the `optimizer` and `scheduler` keys are absent in the DeepSpeed config file.
+
+   c. Custom Optim + DS Scheduler: The case when only `scheduler` key is present in the DeepSpeed config file. 
+   In this situation, user has to use `accelerate.utils.DummyScheduler` to replace the PyTorch/Custom scheduler in their code. 
+
+   d. DS Optim + Custom Scheduler: The case when only `optimizer` key is present in the DeepSpeed config file. 
+   This will result in an error because one can only use DS Scheduler when using DS Optim.
+
+2. Notice the `auto` values in the above example DeepSpeed config files. 
+These have to specified by the user in the code by running `accelerator.state.deepspeed_plugin.deepspeed_config_process(**kwargs)`. 
+This allows user more flexibilty to fill these fields based on the code instead of hardcoding them. 
+Below is the snippet from `examples/by_feature/deepspeed_with_config_support.py` showing how to fill these: 
+```python
```

Comment:
This is a huge block, I'd document less auto option plus make sure we automatically set the ones we can!